src/streamlink/stream/streamprocess.py

```diff
@@ -62,19 +62,22 @@ def is_usable(cls, session):
         raise NotImplementedError
 
     def open(self):
-        process = self.spawn(self.parameters, self.arguments)
+        if self.is_usable(self.session):
+            process = self.spawn(self.parameters, self.arguments)
 
-        # Wait 0.5 seconds to see if program exited prematurely
-        time.sleep(0.5)
+            # Wait 0.5 seconds to see if program exited prematurely
+            time.sleep(0.5)
 
-        if not process.poll() is None:
-            if hasattr(self.stderr, "name"):
-                raise StreamError(("Error while executing subprocess, "
-                                   "error output logged to: {0}").format(self.stderr.name))
-            else:
-                raise StreamError("Error while executing subprocess")
+            if not process.poll() is None:
+                if hasattr(self.stderr, "name"):
+                    raise StreamError(("Error while executing subprocess, "
+                                       "error output logged to: {0}").format(self.stderr.name))
+                else:
+                    raise StreamError("Error while executing subprocess")
 
-        return StreamProcessIO(self.session, process, process.stdout, timeout=self.timeout)
+            return StreamProcessIO(self.session, process, process.stdout, timeout=self.timeout)
+        else:
+            raise StreamError("{0} is not installed or not supported on your system".format(self.shortname()))
```

Comment:
You should probably use the value of the `rtmp-rtmpdump` option here instead of the class's shortname.

Reply:
I would have done, but this is `StreamProcess`.

Reply:
You're right, I was looking at the wrong file locally. You should still be able to use `self.cmd` though if I'm not mistaken, since the implementing classes need to set `self.cmd`.

Reply:
Yep, it's not so well exposed, so I'll refactor it a bit...